src/unix/getaddrinfo.c

```diff
@@ -184,13 +202,18 @@ int uv_getaddrinfo(uv_loop_t* loop,
     req->hostname = memcpy(buf + len, hostname, hostname_len);
 
   if (cb) {
-    uv__work_submit(loop,
-                    &req->work_req,
-                    UV__WORK_SLOW_IO,
-                    uv__getaddrinfo_work,
-                    uv__getaddrinfo_done);
+    /* TODO options should indicate type. */
+    work->data = req;
+    req->reserved[0] = work; /* For uv_cancel. */
```

Comment:
req is a uv_getaddrinfo_t so req->reserved[0] is private from its perspective. Since req->reserved[0] is part of the executor API I thought it would be OK to use it privately here to give a pointer to the uv_work_t it submits.

This enables uv_cancel.

I do the same thing for uv_getnameinfo and uv_fs.